src/vectorDBManager.ts

```diff
@@ -79,56 +109,105 @@ class VectorDBManager {
     }
   }
 
-  public static async getMemoryVectors(docHash: string): Promise<MemoryVector[] | undefined> {
+  public static async loadFile(noteFile: NoteFile, embeddingsAPI: Embeddings) {
     if (!this.db) throw new Error("DB not initialized");
+    const textSplitter = new RecursiveCharacterTextSplitter({ chunkSize: 5000 })
+    const splitDocument = await textSplitter.createDocuments([noteFile.content])
+    const docVectors = await embeddingsAPI.embedDocuments(splitDocument.map((doc) => doc.pageContent))
+    const memoryVectors = docVectors.map((docVector, i) => (
+      {
+        content: splitDocument[i].pageContent,
+        metadata: {
+          ...noteFile.metadata,
+          title: noteFile.basename,
+          path: noteFile.path,
+
+        },
+        embedding: docVector,
+      }))
+    const docHash = VectorDBManager.getDocumentHash(noteFile.path);
```

Comment:
`docHash` as the hash of the note path works well as the id of the entry, but do you think it's also useful to have a `hash(content)` in the entry that can be used to tell whether the note has been modified? Or is it already covered by the mtime mechanism?

Reply:
I think this is already covered by the mtime mechanism. Using the doc content hash has it's merits in terms of being quite exact in the case where a document has been changed and then reverted (with ctrl+z for example) but using mtime allows us to efficiently index files without having to run a hash of all of them.